### PR TITLE
tls: enable allow_expired_certificate for SPIFFE validator

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: tls: enable allow_expired_certificate flag in ValidationContext for SPIFFE validator
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by dio** [comment]
cc. @envoyproxy/first-pass-reviewers

File: source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc [hunk 1]
```diff
@@ -171,6 +175,16 @@ int SPIFFEValidator::doVerifyCertChain(X509_STORE_CTX* store_ctx,
   return ret;
 }
 
+int SPIFFEValidator::ignoreCertificateExpirationCallback(int ok, X509_STORE_CTX* store_ctx) {
```
Comment (asraa) [inline source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc:178]: If this is just duplicated code from https://github.com/envoyproxy/envoy/blob/50e81276fd8f109ba3a6134e790f65c1cc5bdec9/source/extensions/transport_sockets/tls/cert_validator/default_validator.cc#L230
is it possible to just share the code?

File: test/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator_test.cc [hunk 2]
```diff
@@ -397,6 +443,30 @@ name: envoy.tls.cert_validator.spiffe
...
+    X509_STORE_CTX_set_error(store_ctx.get(), X509_V_ERR_CERT_HAS_EXPIRED);
+    EXPECT_TRUE(SPIFFEValidator::ignoreCertificateExpirationCallback(false, store_ctx.get()));
+  }
+  // Yet valid case.
+  {
+    X509StoreContextPtr store_ctx = X509_STORE_CTX_new();
+    X509_STORE_CTX_set_error(store_ctx.get(), X509_V_ERR_CERT_NOT_YET_VALID);
+    EXPECT_TRUE(SPIFFEValidator::ignoreCertificateExpirationCallback(false, store_ctx.get()));
+  }
+  // Other error
```
Comment (asraa) [inline test/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator_test.cc:462]: thanks for testing this case :)

**Review by asraa — COMMENTED** [review]
Thanks!

File: source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc (hunk 1 above)
Comment (mathetake) [reply source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc]: done!

**Review by asraa — APPROVED** [review]
LGTM! @envoyproxy/senior-maintainers for final pass please

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**